packages/pouch-storage/src/PouchCommandHistoryTable.ts

```diff
@@ -3,26 +3,258 @@ import {
   CommandHistoryTable,
 } from '@deephaven/console';
 import { Type as FilterType } from '@deephaven/filters';
-import { StorageUtils } from '@deephaven/storage';
-import PouchStorageTable from './PouchStorageTable';
+import Log from '@deephaven/log';
+import {
+  StorageTableViewport,
+  StorageUtils,
+  ViewportData,
+} from '@deephaven/storage';
+import { PromiseUtils } from '@deephaven/utils';
+import PouchCommandHistoryCache from './PouchCommandHistoryCache';
+import { siftPrunableItems } from './pouchCommandHistoryUtils';
+import PouchStorageTable, {
+  PouchDBSort,
+  PouchStorageItem,
+} from './PouchStorageTable';
+
+const COMMAND_HISTORY_ITEMS_MAX = 2500;
+const COMMAND_HISTORY_ITEMS_PRUNE = 2000;
+
+const log = Log.module('PouchCommandHistoryTable');
+
+type CommandHistoryDoc = PouchDB.Core.ExistingDocument<
+  CommandHistoryStorageItem & PouchStorageItem & PouchDB.Core.AllDocsMeta
+>;
 
 export class PouchCommandHistoryTable
   extends PouchStorageTable<CommandHistoryStorageItem>
   implements CommandHistoryTable {
-  constructor(language: string) {
-    super(`CommandHistoryStorage.${language}`);
-  }
-
-  setSearch(search: string): void {
-    this.setFilters(
-      search
-        ? [
-            StorageUtils.makeFilterConfig([
-              StorageUtils.makeFilterItem('name', FilterType.contains, search),
-            ]),
-          ]
-        : []
+  constructor(language: string, private cache: PouchCommandHistoryCache) {
+    super(`CommandHistoryStorage.${language}`, ({
+      // Optimizations to cut down on growing table size. These should be safe
+      // since we don't care about revision history for command history
+      // documents.
+      auto_compaction: true,
+      revs_limit: 1,
+    } as unknown) as PouchDB.HttpAdapter.HttpAdapterConfiguration);
```

Comment:
We shouldn't be using the `HttpAdapter`, see my other comment on `PouchStorageTable`. No need to cast here.
```suggestion
    super(`CommandHistoryStorage.${language}`, {
      // Optimizations to cut down on growing table size. These should be safe
      // since we don't care about revision history for command history
      // documents.
      auto_compaction: true,
      revs_limit: 1,
    });
```